### PR TITLE
feat: Add 404 page

### LINKED ISSUE #34: feat: Custom 404 Page
Refer: https://www.netlify.com/blog/2020/12/08/making-a-custom-404-page-in-next.js/

### PR DESCRIPTION
**Pull request by Nishith-Savla**

### CODE REVIEW

**Comment by ssarkar551** [comment]
@Nishith-Savla  Try adding screenshots of before and after adding your code. It helps others reviewing it quicker, as we won't have to check for the changes in our local system. This time it's ok, since you already given the screenshot via Whatsapp.

**Review by ssarkar551 — APPROVED** [review]

File: src/pages/404.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import Link from "next/link";
+
+export default function FourOhFour() {
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
const FourOhFour: NextPage = () => {
```

File: src/pages/404.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import Link from "next/link";
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
import Link from "next/link";
import type { NextPage } from "next";
```

File: src/pages/404.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export default function FourOhFour() {
+  return (
+    <main>
+      <div>
+        <div className="h-[calc(100vh-200px)] py-12 sm:py-8 md:py-12 lg:py-14 xl:py-12 2xl:py-28">
+          <div className="flex flex-col items-center">
+            <div className="max-w-md px-4 sm:max-w-2xl sm:px-6 md:max-w-3xl lg:max-w-4xl lg:px-0 xl:max-w-5xl 2xl:max-w-6xl">
+              <h1 className="text-center text-4xl font-bold tracking-tight text-white sm:text-6xl sm:tracking-tight lg:text-[4rem] xl:text-[6rem] xl:tracking-tight 2xl:text-[6.5rem]">
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
              <h1 className="text-center text-4xl font-bold tracking-tight sm:text-6xl sm:tracking-tight lg:text-[4rem] xl:text-[6rem] xl:tracking-tight 2xl:text-[6.5rem]">
```

File: src/pages/404.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                Page not found...
+                <span className="text-[hsl(200,100%,60%)]">404!</span>
+              </h1>
+              <div className="mt-4 flex w-full items-center justify-center gap-4 text-white xl:mt-8">
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
              <div className="mt-4 flex w-full items-center justify-center gap-4 xl:mt-8">
```

File: src/pages/404.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import Link from "next/link";
+
+export default function FourOhFour() {
+  return (
+    <main>
+      <div>
+        <div className="h-[calc(100vh-200px)] py-12 sm:py-8 md:py-12 lg:py-14 xl:py-12 2xl:py-28">
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
        <div className="flex min-h-screen items-center justify-center text-center">
```

File: src/pages/404.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+          <div className="flex flex-col items-center">
+            <div className="max-w-md px-4 sm:max-w-2xl sm:px-6 md:max-w-3xl lg:max-w-4xl lg:px-0 xl:max-w-5xl 2xl:max-w-6xl">
+              <h1 className="text-center text-4xl font-bold tracking-tight text-white sm:text-6xl sm:tracking-tight lg:text-[4rem] xl:text-[6rem] xl:tracking-tight 2xl:text-[6.5rem]">
+                Page not found...
+                <span className="text-[hsl(200,100%,60%)]">404!</span>
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
                <span className="text-[hsl(200,100%,60%)]"> 404!</span>
```

File: src/pages/404.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import Link from "next/link";
+
+export default function FourOhFour() {
+  return (
+    <main>
+      <div>
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx]:
```suggestion
```

File: src/pages/404.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                      className="transition group-hover:translate-x-[3px]"
+                      d="M1 1l4 4-4 4"
+                    ></path>
+                  </svg>
+                </Link>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
```
Comment (AyanavaKarmakar) [inline src/pages/404.tsx:20]:
```suggestion
```

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]

File: src/pages/404.tsx (hunk 3 above)
Comment (Nishith-Savla) [reply src/pages/404.tsx]: This cannot be implemented as black text isn't visible on black bg

Comment (AyanavaKarmakar) [reply src/pages/404.tsx]: @Nishith-Savla the bg ain't black tho (in light theme)

Running this branch in my PC is giving this output

image

image

The global.css file will apply the base styles on all pages.

Comment (Nishith-Savla) [reply src/pages/404.tsx]: Ohh, got it. I didn't have the changes from the dev branch

**Review by AyanavaKarmakar — APPROVED** [review]

**Pull request merged**